internal/controllers/inventory_controller_postgres.go

```diff
@@ -0,0 +1,305 @@
+								Protocol:      corev1.ProtocolTCP,
+								ContainerPort: 5432,
+							},
+						},
+						VolumeMounts: deploymentVolumeMounts,
+					},
+				},
+			},
+		},
+	}
+
+	// Build the deployment.
+	newDeployment := &appsv1.Deployment{
+		ObjectMeta: deploymentMeta,
+		Spec:       deploymentSpec,
+	}
+
+	t.logger.InfoContext(ctx, "[deployDatabase] Create/Update/Patch Server", "Name", serverName)
+	if err := utils.CreateK8sCR(ctx, t.client, newDeployment, t.object, utils.UPDATE); err != nil {
+		return fmt.Errorf("failed to deploy database: %w", err)
+	}
+
+	return nil
+}
+
+// createPasswords sets up the admin and service passwords for the Postgres database.
+func (t *reconcilerTask) createPasswords(ctx context.Context, serverName string) error {
+	// Create the passwords secret
+	passwordSecretName := fmt.Sprintf("%s-passwords", serverName)
+	var existing corev1.Secret
+	err := t.client.Get(ctx, types.NamespacedName{Name: passwordSecretName, Namespace: t.object.Namespace}, &existing)
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("failed to query for existing passwords: %w", err)
+	}
+
+	if errors.IsNotFound(err) {
+		// Does not already exist; create it.
+		passwords := utils.CreateSecretFromLiterals(t.object.Namespace, passwordSecretName, map[string][]byte{
+			utils.AdminPasswordEnvName:     []byte(utils.GetPasswordOrRandom(utils.AdminPasswordEnvName)),
+			utils.AlarmsPasswordEnvName:    []byte(utils.GetPasswordOrRandom(utils.AlarmsPasswordEnvName)),
+			utils.ResourcesPasswordEnvName: []byte(utils.GetPasswordOrRandom(utils.ResourcesPasswordEnvName)),
+		})
+		err = utils.CreateK8sCR(ctx, t.client, passwords, t.object, utils.UPDATE)
+		if err != nil {
+			return fmt.Errorf("failed to create passwords: %w", err)
+		}
+	} else {
+		// Existing passwords exist.  Make sure that required passwords exist.
+		// Only create these once to avoid resetting the passwords.  We could update these passwords, but we would need
+		// to force a restart of the postgres service so that its init scripts run again.  Updating the secret without
+		// restarting the postgres service will cause password authentication to fail for all the servers using the new
+		// set of passwords.
+		if _, ok := existing.Data[utils.AdminPasswordEnvName]; !ok {
+			existing.Data[utils.AdminPasswordEnvName] = []byte(utils.GetPasswordOrRandom(utils.AdminPasswordEnvName))
+		}
+		if _, ok := existing.Data[utils.AlarmsPasswordEnvName]; !ok {
+			existing.Data[utils.AlarmsPasswordEnvName] = []byte(utils.GetPasswordOrRandom(utils.AlarmsPasswordEnvName))
+		}
+		if _, ok := existing.Data[utils.ResourcesPasswordEnvName]; !ok {
+			existing.Data[utils.ResourcesPasswordEnvName] = []byte(utils.GetPasswordOrRandom(utils.ResourcesPasswordEnvName))
+		}
+
+		err = utils.CreateK8sCR(ctx, t.client, &existing, t.object, utils.UPDATE)
+		if err != nil {
+			return fmt.Errorf("failed to create passwords: %w", err)
+		}
+	}
+
+	return nil
```

Comment:
Would nice have some additional success logs and/or debug logs in the funcs (here and others too). Would helpful in prod (and in dev) incase any of the crucial steps fail.

Reply:
added